proposals/NNNN-shader-execution-reordering.md

```diff
@@ -0,0 +1,1388 @@
+
+### New operations
+
+| Opcode | Opcode name | Description
+|:---    |:---         |:---
+XXX      | HitObject_TraceRay | Analogous to TraceRay but without invoking CH/MS and returns the intermediate state as a `HitObject`.
+XXX + 1  | HitObject_FromRayQuery | Creates a new `HitObject` representing a committed hit from a `RayQuery`.
+XXX + 2  | HitObject_FromRayQueryWithAttrs | Creates a new `HitObject` representing a committed hit from a `RayQuery` and committed attributes.
+XXX + 3  | HitObject_MakeMiss | Creates a new `HitObject` representing a miss.
+XXX + 4  | HitObject_MakeNop | Creates an empty nop `HitObject`.
+XXX + 5  | HitObject_Invoke | Represents the invocation of the CH/MS shader represented by the `HitObject`.
+XXX + 6  | ReorderThread | Reorders the current thread. Optionally accepts a `HitObject` arg, or `undef`.
+XXX + 7  | HitObject_IsMiss | Returns `true` if the `HitObject` represents a miss.
+XXX + 8  | HitObject_IsHit | Returns `true` if the `HitObject` represents a hit.
+XXX + 9  | HitObject_IsNop | Returns `true` if the `HitObject` is a Nop-HitObject.
+XXX + 10 | HitObject_RayFlags | Returns the ray flags set in the HitObject.
+XXX + 11 | HitObject_RayTMin | Returns the TMin value set in the HitObject.
+XXX + 12 | HitObject_RayTCurrent | Returns the current T value set in the HitObject.
+XXX + 13 | HitObject_WorldRayOrigin | Returns the ray origin in world space.
+XXX + 14 | HitObject_WorldRayDirection | Returns the ray direction in world space.
+XXX + 15 | HitObject_ObjectRayOrigin | Returns the ray origin in object space.
+XXX + 16 | HitObject_ObjectRayDirection | Returns the ray direction in object space.
+XXX + 17 | HitObject_ObjectToWorld3x4 | Returns the object to world space transformation matrix in 3x4 form.
+XXX + 18 | HitObject_ObjectToWorld4x3 | Returns the object to world space transformation matrix in 4x3 form.
+XXX + 19 | HitObject_WorldToObject3x4 | Returns the world to object space transformation matrix in 3x4 form.
+XXX + 20 | HitObject_WorldToObject4x3 | Returns the world to object space transformation matrix in 4x3 form.
+XXX + 21 | HitObject_GeometryIndex | Returns the geometry index committed on hit.
+XXX + 22 | HitObject_InstanceIndex | Returns the instance index committed on hit.
+XXX + 23 | HitObject_InstanceID | Returns the instance id committed on hit.
+XXX + 24 | HitObject_PrimitiveIndex | Returns the primitive index committed on hit.
+XXX + 25 | HitObject_HitKind | Returns the HitKind of the hit.
+XXX + 26 | HitObject_ShaderTableIndex | Returns the shader table index set for this HitObject.
+XXX + 27 | HitObject_SetShaderTableIndex | Returns a HitObject with updated shader table index.
+XXX + 28 | HitObject_LoadLocalRootTableConstant | Returns the root table constant for this HitObject and offset.
+XXX + 29 | HitObject_Attributes | Returns the attributes set for this HitObject.
```

Comment:
Names are all capitalized here, but lowercase below. I think they should be lowercase.